Patterns/Dutch/Dutch-DateTime.yaml

```diff
@@ -94,25 +94,37 @@ AmPmDescRegex: !nestedRegex
 DescRegex: !nestedRegex
   def: (:?(:?({OclockRegex}\s+)?(?<desc>({AmPmDescRegex}|{AmDescRegex}|{PmDescRegex}|{SpecialDescRegex})))|{OclockRegex})
   references: [ OclockRegex, AmDescRegex, PmDescRegex, AmPmDescRegex, SpecialDescRegex ]
+PmRegex: !nestedRegex
+  def: (?<pm>({ApostrofsRegex}|des)\s+(\bmiddags|avonds|nachts)|((in|tegen|op|om|met)\s+(de\s+)?)(((na)?middag|avond|(midder)?nacht|lunchtijd))|van(avond|nacht)|dag)
+  references: [ ApostrofsRegex ]
+PmRegexFull: !nestedRegex
+  def: (?<pm>(({ApostrofsRegex}|des)\s+(\bmiddags|avonds|nachts)|((in|tegen|op|om|met)\s+(de\s+)?)?(((na)?middag|avond|(midder)?nacht|lunchtijd))|van(avond|nacht)(?!\s+om\b)))
+  references: [ ApostrofsRegex ]
+AmRegex: !nestedRegex
+  def: (?<am>(({ApostrofsRegex}|des)\s+(ochtends|morgens)|((in|tegen|op)\s+de)(\s+(ochtend|morgen))|(?<=gisteren|morgen|vandaag)(ochtend|morgen)|^?ochtend))
+  references: [ ApostrofsRegex ]
+FullDescRegex: !nestedRegex
+  def: ({DescRegex}|{AmRegex}|{PmRegexFull})
+  references: [ DescRegex, AmRegex, PmRegexFull ]
 # Exclude cases that include the "Am/Pm" suffix
 # Exclude when preceded by "$" symbol
 TwoDigitYearRegex: !nestedRegex
-  def: \b(?<!\$)(?<year>([0-9]\d))(?!(\s*((\:\d)|{AmDescRegex}|{PmDescRegex}|\.\d)))\b
+  def: \b(?<!\$)(?<year>([0-24-9]\d))(?!(\s*((\:\d)|\skeer|{AmDescRegex}|{PmDescRegex}|\.\d)))\b
   references: [ AmDescRegex, PmDescRegex]
 YearRegex: !nestedRegex
   def: ({BaseDateTime.FourDigitYearRegex}|{FullTextYearRegex})
   references: [ BaseDateTime.FourDigitYearRegex, FullTextYearRegex ]
 WeekDayRegex: !simpleRegex
   def: \b(?<weekday>((ma|di(ns)?|wo(e(ns)?)?|do|vr(ij)?|za(t)?|zo)[\.\b])|((?:maan|dins|woens|donder|vrij|zater|zon)(dag)?(en)?(middag)?)\b)
 SingleWeekDayRegex: !simpleRegex
-  def: \b(?<weekday>(((ma|di(ns)?|wo(e(ns)?)?|do|vr(ij\.)?|za(t)?|zo)\b(\.)?)|(((?<!in\s+de\s+)(maan|dins|woens|donder|zater|zon)(dag(en)?)?|(?<=op\s+)vrij|vrij(dag(en)?))\b))|zondag|maandag|dinsdag|woensdag|donderdag|vrijdag|zaterdag)
+  def: \b(?<weekday>(((ma|di(ns)?|wo(e(ns)?)?|do|vr(ij\.)?|za(t)?)\b(\.)?)|(((?<!in\s+de\s+)(maan|dins|woens|donder|zater|zon)(dag(en)?)?|(?<=op\s+)vrij|vrij(dag(en)?))\b))|zondag|maandag|dinsdag|woensdag|donderdag|vrijdag|zaterdag)
```

Comment:
_(todo @iMicknl: double check if `zat` is matched now. Zat is also used in spoken / written language and I wonder if we should not just match on `zat.`.)_

https://onzetaal.nl/taaladvies/afkortingen-dagen-en-maanden/

@tellarin would you rather have many false positives on `zat` and zo`, or better to have them not in the recognizer and just match on the abbreviation ending with a dot.

Reply:
@LionbridgeCSII see above. Currently `zat` has a null resolution.

```json
[{
	"text": "zat",
	"start": 0,
	"end": 2,
	"typeName": "datetimeV2.date",
	"resolution": null
}]
```

```json
[{
	"text": "zat.",
	"start": 0,
	"end": 3,
	"typeName": "datetimeV2.date",
	"resolution": {
		"values": [{
			"timex": "XXXX-WXX-6",
			"type": "date",
			"value": "2021-06-05"
		}, {
			"timex": "XXXX-WXX-6",
			"type": "date",
			"value": "2021-06-12"
		}]
	}
}]
```

Reply:
fixed resolution of "zat" without dot. 
 Since "zat|zon?|vrij" are ambiguous, they are now extracted (as isolated dates) only if followed by a dot (and not at the end of the sentence) or if preceded by "op|voor". 
Test cases added.

Reply:
I would say that abbreviations without a dot like zat/zo/zon/vrij are not commonly used. If I am missing a use-case, we would know soon enough and it shouldn't be too much work. However I would say that the amount of false positives will hurt more if we enable them broadly.

Current solution described by @LionbridgeCSII above sounds good.